app/src/main/java/pl/org/akai/eater/contracts/AuthResultContract.kt

```diff
@@ -0,0 +1,22 @@
+package pl.org.akai.eater.contracts
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.activity.result.contract.ActivityResultContract
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.tasks.Task
+import pl.org.akai.eater.util.getGoogleSignInClient
+
+class AuthResultContract : ActivityResultContract<Int, Task<GoogleSignInAccount>?>() {
+    override fun createIntent(context: Context, input: Int?): Intent =
+        getGoogleSignInClient(context).signInIntent.putExtra("input", input)
+
+    override fun parseResult(resultCode: Int, intent: Intent?): Task<GoogleSignInAccount>? {
+        return when(resultCode) {
+            Activity.RESULT_OK -> GoogleSignIn.getSignedInAccountFromIntent(intent)
```

Comment:
I get RESULT_CANCELED here, it would be great if to handle also that case, because I didn't get any message.
Also it is tstrange that i get RESULT_CANCELED, becaouse i do not cancel logging flow, but pick one of my accounts.

Reply:
This one is strange and I got some troubles reproducing it. When I have previously selected account for this app I get result code "OK". When I have to select one of my google accounts (fresh install/cleared cache) I still pass to recipes list but function parseResult isn't invoked in this case - couldn't log actual value of result code. Guess this is handled by GoogleSignInAccount.